src/components/views/messages/MessageActionBar.tsx

```diff
@@ -67,8 +67,9 @@ const OptionsButton: React.FC<IOptionsButtonProps> = ({
     const [menuDisplayed, button, openMenu, closeMenu] = useContextMenu();
     const [onFocus, isActive, ref] = useRovingTabIndex(button);
     useEffect(() => {
+        onFocus();
         onFocusChange(menuDisplayed);
-    }, [onFocusChange, menuDisplayed]);
+    }, [onFocus, onFocusChange, menuDisplayed]);
```

Comment:
Trying to get more knowledgeable around the `RovingTabIndex` stuff. Why do we want to call the `onFocus` callback which sets its focus when the component mounts?

We also do the same thing with `onFocusChange` below on line 116. I assume we want to make the same change there?

Reply:
So this is because the button's onFocus doesn't fire when it is pressed (mouse) as its click handler pre-empts it, so without this change when you close the menu you'll roving-tabindex back to some random other button in the message action bar, rather than the one which owns the context menu.

> Trying to get more knowledgeable around the RovingTabIndex stuff. Why do we want to immediately call the onFocus callback which sets its focus after the component mounts?

> We also do the same thing with onFocusChange below on line 116. I assume we want to make the same change there?

Yup, that seems right, probably didn't hit this issue because its the first button so is the default roving-tabindex within the widget

Reply:
Done

Reply:
Can we `SetFocus` in the internals of `useRovingTabIndex` where we have a `useLayoutEffect`? [`src/accessibility/RovingTabIndex.tsx#L257-L279`](https://github.com/matrix-org/matrix-react-sdk/blob/132e19beb77d89d70c08c9d831f950bb0019d7d6/src/accessibility/RovingTabIndex.tsx#L257-L279)

Otherwise, would be good to have a comment wherever we do this as it's not straightforward to connect this benefit

Reply:
`useLayoutEffect` happens on each render, not on click/focus though?

Reply:
Can we use `useEffect` ? It seems like we can do the same `useEffect` but in `useRovingTabIndex` so it happens automatically.

Are there cases where we don't want this to happen?

Reply:
The actual trigger that is making this work is menuDiplayed changing, useRoving doesn't have access to that state

The other two are in the deps just to honor the rule of hooks in exhaustive dependencies

Now that I think about it, they arguably shouldn't be there and this approach is bad as it'll affect the default roving tabindex. Need to just hit onFocus when calling showMenu

Reply:
> The actual trigger that is making this work is menuDiplayed changing, useRoving doesn't have access to that state

Ahhh, I think adding it to our `onClick` handler will make the connection more clear too 👌